### PR TITLE
JBIDE-25908 - allow setup-cdk to be run more conveniently during launch

### PR DESCRIPTION
**Pull request by robstryker**
Signed-off-by: Rob Stryker <rob@oxbeef.net>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by odockal** [comment]
@robstryker Thanks for incorporating patch so fast!

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/controllers/CDK3LaunchController.java [hunk 1]
```diff
@@ -210,8 +214,76 @@ public void launch(ILaunchConfiguration configuration, String mode, ILaunch laun
...
+	}
+	
+	private void launchCheckSetupCDK(IServer s, CDKServer cdkServer, ControllableServerBehavior beh) throws CoreException {
+		CDK3Server cdk3 = (CDK3Server)cdkServer;
+		if( !cdk3.isCDKInitialized()) {
+			int[] retmain = new int[1];
+			retmain[0] = -1;
+			String home = cdk3.getMinishiftHome();
+			Display.getDefault().syncExec(() -> {
+				MessageBox messageBox = new MessageBox(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell(), SWT.ICON_WARNING | SWT.OK | SWT.CANCEL);
```
Comment (odockal) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/controllers/CDK3LaunchController.java]: Please, change MessageBox to MessageDialog: new MessageDialog(shell, title, null, msg, MessageDialog.WARNING, new String[] {"OK", "Cancel"}, 0);

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/controllers/CDK3LaunchController.java [hunk 2]
```diff
@@ -210,8 +214,76 @@ public void launch(ILaunchConfiguration configuration, String mode, ILaunch laun
...
+				String msgText = "Your CDK installation has not been properly initialized. Would you like us to run setup-cdk for you?";
+				String msg = NLS.bind(msgText, home);
+				messageBox.setMessage(msg);
+				messageBox.setText("Warning: CDK has not been properly initialized!");
+				retmain[0] = messageBox.open();
+			});
+			if( retmain[0] == SWT.OK) {
```
Comment (odockal) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/controllers/CDK3LaunchController.java]: SWT.OK -> IDialogConstants.OK_ID

**Comment by adietish** [comment]
discussed things with rob, seems good to me given that he'll change some of the complaints. +1

**Comment by jkopriva** [comment]
@robstryker btw: copyright and javadoc are missing in file SetupCDKJob.java.

**Pull request merged**